nsqadmin/nsqadmin.go

```diff
@@ -130,11 +131,28 @@ func New(opts *Options) *NSQAdmin {
 		}
 	}
 
+	opts.BasePath = normalizeBasePath(path.Clean(opts.BasePath))
+
 	n.logf(LOG_INFO, version.String("nsqadmin"))
 
 	return n
 }
 
+func normalizeBasePath(p string) string {
+	if len(p) == 0 {
+		return "/"
+	}
+	// add leading slash
+	if p[0] != '/' {
+		p = "/" + p
+	}
+	// add trailing slash
+	if p[len(p)-1] != '/' {
+		p = p + "/"
```

Comment:
I think the templated urls and such may look a bit better if instead `BasePath` does _not_ end in `/` and we write the `/` after it everywhere. But I understand if you disagree, and it's OK this way too.

Reply:
Well, except for `/`. I got this from Backbone's code and was trying to create some parity.

I do agree that aesthetically, the sites where we use `basePath` would look better with a preceding `/`. But really it's all just fugly.

Reply:
We should make a decision on this...

Reply:
I've noted my preference, and I leave the choice up to you :)

Reply:
(I think `BasePath` and `base_path` could be blank in the default case, and the root path could be written as `{{BasePath}}/` and `{{base_path}}/`)

Reply:
(I mean, in the config as `"/"`, but in this function turned into `""` by removing the trailing (and only) slash)